pkg/web/client.go

```diff
@@ -0,0 +1,239 @@
+package web
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/replicate/cog/pkg/config"
+	"github.com/replicate/cog/pkg/docker/command"
+	"github.com/replicate/cog/pkg/env"
+	"github.com/replicate/cog/pkg/global"
+)
+
+type Client struct {
+	command command.Command
+	client  *http.Client
+}
+
+type File struct {
+	Path   string `json:"path"`
+	Digest string `json:"digest"`
+	Size   int64  `json:"size"`
+}
+
+type Env struct {
+	CogGpu              string `json:"COG_GPU"`
+	CogPredictTypeStub  string `json:"COG_PREDICT_TYPE_STUB"`
+	CogTrainTypeStub    string `json:"COG_TRAIN_TYPE_STUB"`
+	CogPredictCodeStrip string `json:"COG_PREDICT_CODE_STRIP"`
+	CogTrainCodeStrip   string `json:"COG_TRAIN_CODE_STRIP"`
+	CogPyEnvPath        string `json:"COG_PYENV_PATH"`
+	CogEagerImports     string `json:"COG_EAGER_IMPORTS"`
+	R8CudaVersion       string `json:"R8_CUDA_VERSION"`
+	R8CudnnVersion      string `json:"R8_CUDNN_VERSION"`
+	R8PythonVersion     string `json:"R8_PYTHON_VERSION"`
+	R8TorchVersion      string `json:"R8_TORCH_VERSION"`
+}
+
+type RuntimeConfig struct {
+	Weights []File `json:"weights"`
+	Files   []File `json:"files"`
+	Env     Env    `json:"env"`
+}
+
+type Version struct {
+	Annotations   map[string]string      `json:"annotations"`
+	CogConfig     config.Config          `json:"cog_config"`
+	CogVersion    string                 `json:"cog_version"`
+	OpenAPIScheme map[string]interface{} `json:"openapi_schema"`
+	RuntimeConfig RuntimeConfig          `json:"runtime_config"`
+	Virtual       bool                   `json:"virtual"`
+}
+
+func NewClient(command command.Command, client *http.Client) *Client {
```

Comment:
nit: I'd change the local `command` to something else so that it's not shadowing the imported package name

Reply:
Resolved in [5ee1230](https://github.com/replicate/cog/pull/2139/commits/5ee1230d0507082d8be36be2a2973c06bd46680e)